R-package/R/lgb.Booster.R

```diff
@@ -837,6 +838,10 @@ Booster <- R6::R6Class(
 #'
 #'             Note that, if using custom objectives, types "class" and "response" will not be available and will
 #'             default towards using "raw" instead.
+#'
+#'             If the model was fit through function \link{lightgbm}, passing the prediction type through \code{params}
+#'             instead of through this argument might result in factor levels (for classification objectives) not being
+#'             applied correctly to the resulting output.
```

Comment:
```suggestion
#'             If the model was fit through function \link{lightgbm} and it was passed a factor as labels,
#'             passing the prediction type through \code{params} instead of through this argument might
#'             result in factor levels for classification objectives not being applied correctly to the resulting output.
```

Thanks for drawing attention to this in the PR description and this note. Since this is new functionality, I think it's ok for now to just have a warning in docs and to maybe find a way to resolve this inconsistency in the future.

But please consider this suggestion to make it clear that this warning is specific to the situation where a factor was provided for the label.